projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -827,9 +827,11 @@ export class DatagridComponent<R extends B, B = any> implements OnInit, AfterVie
         if (this.datagrid.selection.currentSingle) {
```

Comment:
Shouldn't we change this one also?
if (this.datagrid.selection.currentSingle) {
   if (this._datagridSelection?.length !== 1 || this._datagridSelection[0] !== this.datagrid.selection.currentSingle) {
      this._datagridSelection = [this.datagrid.selection.currentSingle];
   }
}

Reply:
👍